Storage/container/one_to_one.h

```diff
@@ -108,47 +115,55 @@ class GenericOne2One {
     if (it != forward_.end()) {
-      const T prev_object = *(it->second);
-      const auto row_col = std::make_pair(row, sfinae::GetCol(prev_object));
-      journal_.LogMutation(T_DELETE_EVENT(prev_object),
-                           [this, row_col, prev_object]() { DoAdd(row_col, prev_object); });
-      DoErase(row_col);
+      const T previous_object = *(it->second);
+      const auto key = std::make_pair(row, sfinae::GetCol(previous_object));
+      journal_.LogMutation(T_DELETE_EVENT(previous_object),
+                           [this, key, previous_object]() { DoAdd(key, previous_object); });
+      DoErase(key);
     }
   }
 
   void EraseCol(sfinae::CF<T_COL> col) {
     const auto it = transposed_.find(col);
     if (it != transposed_.end()) {
-      const T prev_object = *(it->second);
-      const auto row_col = std::make_pair(sfinae::GetRow(prev_object), col);
-      journal_.LogMutation(T_DELETE_EVENT(prev_object),
-                           [this, row_col, prev_object]() { DoAdd(row_col, prev_object); });
-      DoErase(row_col);
+      const T previous_object = *(it->second);
+      const auto key = std::make_pair(sfinae::GetRow(previous_object), col);
+      journal_.LogMutation(T_DELETE_EVENT(previous_object),
+                           [this, key, previous_object]() { DoAdd(key, previous_object); });
+      DoErase(key);
     }
   }
 
-  ImmutableOptional<T> operator[](const T_RCPAIR& row_col) const {
-    const auto it = map_.find(row_col);
-    if (it == map_.end()) return nullptr;
-    return ImmutableOptional<T>(FromBarePointer(), it->second.get());
+  ImmutableOptional<T> operator[](const T_KEY& key) const {
+    const auto it = map_.find(key);
+    if (it != map_.end()) {
+      return ImmutableOptional<T>(FromBarePointer(), it->second.get());
+    } else {
+      return nullptr;
+    }
   }
   ImmutableOptional<T> Get(sfinae::CF<T_ROW> row, sfinae::CF<T_COL> col) const {
     return operator[](std::make_pair(row, col));
   }
   ImmutableOptional<T> GetRow(sfinae::CF<T_ROW> row) const {
     const auto it = forward_.find(row);
-    if (it == forward_.end()) return nullptr;
-    return ImmutableOptional<T>(FromBarePointer(), it->second);
+    if (it != forward_.end()) {
+      return ImmutableOptional<T>(FromBarePointer(), it->second);
+    } else {
+      return nullptr;
+    }
   }
   ImmutableOptional<T> GetCol(sfinae::CF<T_COL> col) const {
     const auto it = transposed_.find(col);
-    if (it == transposed_.end()) return nullptr;
-    return ImmutableOptional<T>(FromBarePointer(), it->second);
+    if (it != transposed_.end()) {
+      return ImmutableOptional<T>(FromBarePointer(), it->second);
+    } else {
+      return nullptr;
+    }
   }
 
-  bool CanAdd(const T_RCPAIR& row_col) const {
-    return map_.find(row_col) == map_.end() && forward_.find(row_col.first) == forward_.end() &&
-           transposed_.find(row_col.second) == transposed_.end();
+  bool CanAdd(const T_KEY& key) const {
```

Comment:
Good point. And I don't like `GetRow` and `GetCol` names either. They sound like the methods do directly the opposite. May be, `GetRowEntry` and `GetColEntry` would be better. What do you think?